modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/fragment/renderer/FDSAdminFragmentRenderer.java

```diff
@@ -647,6 +652,19 @@ private JSONArray _getFiltersJSONArray(
 						"type", "selection"
 					);
 
+					if (Objects.equals(sourceType, "API_REST_APPLICATION")) {
+						selectionFilterJSONObject.put("id", fieldName);
+					}
+					else {
+						selectionFilterJSONObject.put(
```

Comment:
Filters of type `PICKLIST` are supposed to work with string nodes, but after testing we need to remove the `key/name` part to make it work.

![image-20240911-111118](https://github.com/user-attachments/assets/2e27e131-be6b-462a-9638-06f9f26ee059)

In this case, the filter works using the expression `universityStudents/studentPicklist`, so we need to remove the selected leaf part.